### PR TITLE
Plugin for text embedding

### PR DESCRIPTION
**Pull request by ddannyc**
### Background
This pull request duplicates [#2804](https://github.com/Significant-Gravitas/Auto-GPT/pull/2804), but that pull request did not pass the tests and may not be merged.

### Changes
* autogpt/llm_utils.py#create_embedding  check and handle embedding
* autogpt/models/base_open_ai_plugin.py add embedding plugin

### Test Plan

tests/unit/test_plugins.py::test_scan_plugins_openai

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/models/base_open_ai_plugin.py [hunk 1]
```diff
@@ -197,3 +197,21 @@ def handle_chat_completion(
...
+        """This method is called to check that the plugin can
+          handle the text_embedding method.
+        Args:
+            text (str): The text to be convert to embedding.
+          Returns:
+              bool: True if the plugin can handle the text_embedding method."""
+        return False
+
+    def handle_text_embedding(self, text: str, *_, **kwargs) -> list:
+        """This method is called when the chat completion is done.
```
Comment (ntindle) [inline autogpt/models/base_open_ai_plugin.py]: Doc string is incorrect here

File: autogpt/llm/llm_utils.py [hunk 2]
```diff
@@ -251,6 +251,12 @@ def create_embedding(
         openai.Embedding: The embedding object.
     """
     cfg = Config()
+    for plugin in cfg.plugins:
+        if plugin.can_handle_text_embedding(text, **kwargs):
+            embedding = plugin.handle_text_embedding(text, **kwargs)
+            if embedding is not None:
+                return embedding
```
Comment (ntindle) [inline autogpt/llm/llm_utils.py]: Can multiple embeddings work here? "No, that's annoying is an acceptable answer"

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt/llm/llm_utils.py (hunk 2 above)
Comment (ddannyc) [reply autogpt/llm/llm_utils.py]: The embedding plugin works like chat completion; for now, it only accepts a one plugin.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
Closing this as a duplicate of #2804. Still, thanks for picking it up! Turns out, it was just the linter complaining on that PR, so would have been easy to fix. Now it took a little bit more work to get it up to date with `master`, but still fairly easy to fix up & merge. :)

**Pull request closed without merging**